pkgs/development/haskell-modules/generic-builder.nix

```
@@ -802,7 +802,7 @@ stdenv.mkDerivation ({
         buildInputs =
           otherBuildInputsSystem;
         LANG = "en_US.UTF-8";
-        LOCALE_ARCHIVE = lib.optionalString (stdenv.hostPlatform.libc == "glibc") "${buildPackages.glibcLocales}/lib/locale/locale-archive";
+        LOCALE_ARCHIVE = lib.optionalString (stdenv.hostPlatform.libc == "glibc" && buildPackages.glibcLocales != null) "${buildPackages.glibcLocales}/lib/locale/locale-archive";
```

```suggestion
        LOCALE_ARCHIVE = lib.optionalString (stdenv.buildPlatform.libc == "glibc") "${buildPackages.glibcLocales}/lib/locale/locale-archive";
```

This would make the condition the same we use in the actual builder, probably old code that never got fixed. `LOCALE_ARCHIVE` only influences the runtime so it makes no sense to set it based on host in the build environment.

https://github.com/NixOS/nixpkgs/blob/e1bdae5eb8c5efd0da2ba8882c257b4a73137a58/pkgs/development/haskell-modules/generic-builder.nix#L855